internal/pkg/controllers/tunnel.go

```diff
@@ -177,232 +198,37 @@ func (tc *TunnelController) Create(ctx context.Context, params *options.Paramete
 	if err != nil {
 		return err
 	}
-
+	// Map the API response to the struct
 	tunnelCreated := tunResp.Data
+	// Enrich with local data
 	tunnelCreated.RportServer = tc.Rport.BaseURL
-	tunnelCreated.Usage = tc.generateUsage(tunnelCreated, params)
-	if tunnelCreated.ClientID == "" {
-		tunnelCreated.ClientID = clientID
-	}
-	if tunnelCreated.ClientName == "" && clientName != "" {
-		tunnelCreated.ClientID = clientName
-	}
-
-	if clientName == "" && tunnelCreated.ClientName != "" {
-		clientName = tunnelCreated.ClientName
-	}
+	tunnelCreated.ClientID = clientID
```

Comment:
ClientName is cleanly resolved by an API request before creating the tunnel.
`clientID, clientName, err := tc.getClientIDAndClientName(ctx, params)` 
We can trust this value because on error the tunnel creation is aborted.

Reply:
@thorstenkramm if `clientID` is not empty, then `getClientIDAndClientName` returns early with empty `clientName`, only when `clientName` is specified it uses the api to resolve the `client.ID`.